src/components/QueryDrawer.tsx

```diff
@@ -40,46 +40,35 @@ const SavedQuery = ({
   uid,
   isSavedToPage = false,
   onDelete,
-  resultsReferenced,
   clearOnClick,
-  setResultsReferenced,
   editSavedQuery,
   initialResults,
-  onRefresh,
 }: {
   uid: string;
   onDelete?: () => void;
   isSavedToPage?: boolean;
-  resultsReferenced: Set<string>;
   clearOnClick: (s: string) => void;
-  setResultsReferenced: (s: Set<string>) => void;
   editSavedQuery: (s: string) => void;
   initialResults?: Result[];
-  onRefresh: () => void;
 }) => {
   const [results, setResults] = useState<Result[]>(initialResults || []);
-  const resultFilter = useCallback(
-    (r: Result) => !resultsReferenced.has(r.text),
-    [resultsReferenced]
-  );
```

Comment:
A big part of these changes was trying to get rid of custom logic that query editor/results view had to handled.

Here, `resultsReferenced` could now be handled with the `is in canvas` and `is referenced by block in page with title` conditions